### ISSUE TITLE
Local storage should use a random storage key unless unique key is defined.

### ISSUE THREAD

**Issue by TriangleJuice** [digipolisantwerp/acpaas-ui_angular#200]
## Describe the bug
When two apps use the navigation menu and there's no unique storage key specified in both apps, the one app will inherit the state of the other and vice-versa. This can be troublesome if one app uses the docked the mode and the other doesn't

## To Reproduce

1. Integrate navigation menu in app A with possibility to switch to docked mode
2. Integrate navigation menu in app B with NO possibility to switch to docked mode
3. Go to app A and activate docked mode
4. Go to app B and reload

## Expected behavior

App B shouldn't inherit the local storage preferences of app B.

## Screenshots

<img width="445" alt="Screenshot 2020-06-09 at 16 57 07" src="https://user-images.githubusercontent.com/2149758/84164255-97eb5f00-aa72-11ea-8ed9-565e80ff2253.png">

## OS (please complete the following information):

N/A

## Additional context

N/A

**Comment by jsebrech** [comment]
Localstorage is specific to a browser origin, so I don't really see how you can get this in production (on localhost it is indeed quite likely).

**Comment by TriangleJuice** [comment]
Because - if I'm right - on production local storage will not make a difference between the many subdomains.

**Comment by jsebrech** [comment]
No, subdomains are still separate origins, see https://stackoverflow.com/questions/4026479/use-localstorage-across-subdomains

**Comment by TriangleJuice** [comment]
I suggest that we do some research on this and then close the issue if you're right. Thanks for starting the conversation :-)
